### ISSUE TITLE
Rework and unification of the data_provider Python tests 

### ISSUE THREAD

**Issue by pereyra-m** [wazuh/wazuh#17391]
### Description

After the issues #16523 and #3859 are merged, will end up having many similar tests in the folder `src/data_provider/qa`. 
We could instead have only one python script that compares the test tool output against all the required schemas.
This will make the tests easier to understand and maintain.

### DoD

- [ ] Create only one Python script that uses fixtures
- [ ] Every schema validation should be a different test, but the call to the test tool should be done only once
- [ ] CI Pass

**Comment by vikman90** [comment]
Closing this issue. Superseded by version 5.0.
